### PR TITLE
Added support for handling chaining of Symbolic operators and printing without assignment .

### PR DESCRIPTION
**Pull request by anutosh491**
This Pr starts small . For eg
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
from sympy import pi
def main0():
    print(pi + pi)

main0()

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --show-c examples/expr2.py 

// Implementations
void main0()
{
    basic queue0;
    basic_new_stack(queue0);
    basic queue1;
    basic_new_stack(queue1);
    basic_const_pi(queue1);
    basic queue2;
    basic_new_stack(queue2);
    basic_const_pi(queue2);
    basic_add(queue0, queue1, queue2);
    printf("%s\n", basic_str(queue0));
}

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --backend=c --enable-symengine examples/expr2.py 
2*pi
```

**But once we have a queue/stack data structure (I've chosen a queue as it is more intuitive at places) in place to generate temporary variables , we can essentially do much more as shown in the comment below.**

### CODE REVIEW

**Comment by anutosh491** [comment]
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
from sympy import Symbol, pi
def main0():
    x: S = Symbol('x')
    y: S = Symbol('y')
    z: S = (x - S(20)) * (pi + y + S(40))
    x = z
    y = x
    z = y + pi
    print(z)

main0()

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --show-c examples/expr2.py 

// Implementations
void main0()
{
    basic x;
    basic_new_stack(x);
    basic y;
    basic_new_stack(y);
    basic z;
    basic_new_stack(z);
    symbol_set(x, "x");
    symbol_set(y, "y");
    basic queue0;
    basic_new_stack(queue0);
    basic queue1;
    basic_new_stack(queue1);
    integer_set_si(queue1, 20);
    basic_sub(queue0, x, queue1);
    basic queue2;
    basic_new_stack(queue2);
    basic queue3;
    basic_new_stack(queue3);
    basic queue4;
    basic_new_stack(queue4);
    basic_const_pi(queue4);
    basic_add(queue3, queue4, y);
    basic queue5;
    basic_new_stack(queue5);
    integer_set_si(queue5, 40);
    basic_add(queue2, queue3, queue5);
    basic_mul(z, queue0, queue2);
    basic_assign(x, z);
    basic_assign(y, x);
    basic queue6;
    basic_new_stack(queue6);
    basic_const_pi(queue6);
    basic_add(z, y, queue6);
    printf("%s\n", basic_str(z));
}

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --backend=c --enable-symengine examples/expr2.py 
(-20 + x)*(40 + y + pi) + pi
```

**Comment by anutosh491** [comment]
@certik I shall do some more refactoring and explain you the code (mostly the approach) tomorrow in our weekly meeting . I did good amount of testing with random examples and everything works , if you have something particular to test let me know .

File: src/libasr/codegen/asr_to_c.cpp [hunk 1]
```diff
@@ -1138,6 +1138,10 @@ R"(    // Initialise Numpy
             if( ASRUtils::is_array(value_type) ) {
                 src += "->data";
             }
+            if (value_type->type == ASR::ttypeType::SymbolicExpression) {
```
[the remark this answers is not in the document]
Comment (anutosh491) [reply src/libasr/codegen/asr_to_c.cpp]: The whole if else ladder there was using the similar construct as I was doing . I ended up changing all of those to use `ASR::is_a<>()` . Hope that works !

**Comment by anutosh491** [comment]
I've added a test . It might look a bit random but I guess that's the only way we can check all edge cases for operator chaining along with brackets placed randomly . Though I've compared all tests with SymPy and everything looks good 

```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
from sympy import Symbol, pi, S
#from lpython import S

def test_chained_operations():
    x: S = Symbol('x')
    y: S = Symbol('y')
    z: S = Symbol('z')
    a: S = Symbol('a')
    b: S = Symbol('b')
    
    # Chained Operations
    w: S = (x + y) * ((a - b) / (pi + z))
    result: S = (w ** S(2) - pi) + S(3)
    
    # Print Statements
    print("Result 1:", result)
    
    # Additional Operations
    c: S = Symbol('c')
    d: S = Symbol('d')
    e: S = Symbol('e')
    f: S = Symbol('f')
    
    # Chained Operations with Additional Variables
    x = (c * d + e) / f
    y = (x - S(10)) * (pi + S(5))
    z = y ** (S(2) / (f + d))
    result = (z + e) * (a - b)
    
    # Print Statements
    print("Result 2:", result)
    print("x:", x)
    print("y:", y)
    print("z:", z)
    
test_chained_operations()
```
1) Results through sympy 
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ python examples/expr2.py 
Result 1: (a - b)**2*(x + y)**2/(z + pi)**2 - pi + 3
Result 2: (a - b)*(e + ((-10 + (c*d + e)/f)*(pi + 5))**(2/(d + f)))
x: (c*d + e)/f
y: (-10 + (c*d + e)/f)*(pi + 5)
z: ((-10 + (c*d + e)/f)*(pi + 5))**(2/(d + f))
```
2) Results through LPython
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython --backend=c --enable-symengine examples/expr2.py 
Result 1: 3 + (a - b)**2*(x + y)**2/(z + pi)**2 - pi
Result 2: (a - b)*(e + ((5 + pi)*(-10 + (e + c*d)/f))**(2/(d + f)))
x: (e + c*d)/f
y: (5 + pi)*(-10 + (e + c*d)/f)
z: ((5 + pi)*(-10 + (e + c*d)/f))**(2/(d + f))
```

**Comment by anutosh491** [comment]
Actually we might see that there are some differences between how stuff is being printed 
`Result 1: (a - b)**2*(x + y)**2/(z + pi)**2 - pi + 3` from sympy
vs 
`Result 1: 3 + (a - b)**2*(x + y)**2/(z + pi)**2 - pi` from Lpython

I am not sure why this would happen, maybe some internal differences between how sympy and symengine's C wrapper have been framed . Because the C code generated is absolutely spot on . For eg I'll paste the last few lines for C code for `result: S = (w ** S(2) - pi) + S(3))` from Lpython.

```
    basic queue22;
    basic_new_stack(queue22);
    integer_set_si(queue22, 3);
    basic_add(result, queue17, queue22);
    printf("%s\n", basic_str(result));
```

As you see `S(3)` is being added right associatively which is also being shown through `basic_add(result, queue17, queue22)`. But when we link lsymengine and get a result out of it , the 3 is shifted at the front 
`3 + (a - b)**2*(x + y)**2/(z + pi)**2 - pi`

I don't think this might be concerning unless and untill the expressions are same mathematically , though I thought of pointing it out if ever thought why something like this might be happening .
